src/NuGet.Core/NuGet.Build.Tasks.Pack/IPackTaskLogic.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using NuGet.Commands;
+using NuGet.Packaging;
+
+namespace NuGet.Build.Tasks.Pack
+{
+    /// <summary>
+    /// The logic for converting the set of pack MSBuild task parameters to a fully initialized package builder. The
+    /// set of parameters provided to the MSBuild pask task is <see cref="IPackTaskRequest{TItem}"/>. This interface
+    /// allows the logic of the pack task to be seperated from the MSBuild-specific types. The motivation is
+    /// testability.
+    /// </summary>
+    public interface IPackTaskLogic
```

Comment:
Just a side note, it resembles the GoF Strategy pattern.
> Define a family of algorithms, encapsulate each one, and make them interchangeable. Strategy lets the algorithm vary independently from clients that use it.

Perhaps we can follow the same design for other task implementations.